src/source/Signaling/Signaling.c

```diff
@@ -203,10 +203,14 @@ STATUS freeSignaling(PSignalingClient* ppSignalingClient)
     terminateOngoingOperations(pSignalingClient, TRUE);
 
     if (pSignalingClient->pLwsContext != NULL) {
-        MUTEX_LOCK(pSignalingClient->lwsServiceLock);
+        if(IS_VALID_MUTEX_VALUE(pSignalingClient->lwsServiceLock)) {
```

Comment:
If the pLwsContext != NULL then the mutex is not null

Reply:
But https://github.com/awslabs/amazon-kinesis-video-streams-webrtc-sdk-c/blob/b9e41d1c114acd50da3e72b05d486ebb1a0d2ff7/src/source/Signaling/Signaling.c#L100 `pLwsContext` is created before `lwsServiceLock`, may I know how can we guarantee mutex is not null when `(pSignalingClient->pLwsContext != NULL)`

Reply:
You are absolutely correct and I am wrong. I confused with the lws call object we create perform the actual les calls. I think a cleaner solution perhaps is to move the lws context creation below the mutex and other object creations. Just above the stepping of the signaling state machine. What do you think?

Reply:
Move the lws context creation sounds more reasonable than this commit, I will update it, thanks

Reply:
ready to merge, please review again.